ideascube/monitoring/forms.py

```diff
@@ -143,14 +155,13 @@ class ReturnForm(forms.Form):
 
     def clean_loan(self):
         barcode = self.cleaned_data['loan']
-        try:
-            loan = Loan.objects.due().get(specimen__barcode=barcode)
-        except Loan.DoesNotExist:
-            msg = _('Item with barcode {barcode} is not loaned.')
+        loan = Loan.objects.due().filter(specimen__barcode=barcode).first()
+        if loan is None:
+            loan = Loan.objects.due().filter(specimen__serial=barcode).first()
+        if loan is None:
+            msg = _('Item with barcode or serial number {barcode} is not '
+                    'loaned.')
```

Comment:
> Because this (previous) `try`/`except` except for `Loan.DoesNotExist` but also `Loan.MultipleObjectsReturned`.

Right, but given the comment I think we could just remove that case, provided all boxes are upgraded past that version. (see my next comment)